lightning/src/ln/splicing_tests.rs

```diff
@@ -157,3 +154,42 @@ fn test_v1_splice_in() {
 		acceptor_node.node.get_our_node_id()
 	);
 }
+
+#[test]
+fn test_v1_splice_in_negative_insufficient_inputs() {
+	use crate::util::errors::APIError;
+
+	let chanmon_cfgs = create_chanmon_cfgs(2);
+	let node_cfgs = create_node_cfgs(2, &chanmon_cfgs);
+	let node_chanmgrs = create_node_chanmgrs(2, &node_cfgs, &[None, None]);
+	let nodes = create_network(2, &node_cfgs, &node_chanmgrs);
+
+	let (_, _, channel_id, _) =
+		create_announced_chan_between_nodes_with_value(&nodes, 0, 1, 100_000, 0);
+
+	// Amount being added to the channel through the splice-in
+	let splice_in_sats = 20_000;
+
+	// Create additional inputs, but insufficient
+	let extra_splice_funding_input_sats = 19_900;
+	assert!(extra_splice_funding_input_sats <= splice_in_sats);
+	let funding_inputs =
+		create_dual_funding_utxos_with_prev_txs(&nodes[0], &[extra_splice_funding_input_sats]);
+
+	// Initiate splice-in, with insufficient input contribution
+	let res = nodes[0].node.splice_channel(
+		&channel_id,
+		&nodes[1].node.get_our_node_id(),
+		splice_in_sats as i64,
+		funding_inputs,
+		1024, // funding_feerate_per_kw,
+		None, // locktime
+	);
+	assert!(res.is_err());
+	match res {
+		Err(APIError::APIMisuseError { err }) => {
+			assert!(err.contains("Insufficient inputs for splicing"))
+		},
+		_ => panic!("Wrong error {:?}", res.err().unwrap()),
+	}
```

Comment:
Nit: Since we're already matching on the exact error in the next line, maybe we can skip the `res.is_err()` check.

```suggestion
	match res {
		Err(APIError::APIMisuseError { err }) => {
			assert!(err.contains("Insufficient inputs for splicing"))
		},
		_ => panic!("Unexpected result"),
	}
```